doc/OL_preflight_check.py

```diff
@@ -0,0 +1,358 @@
+    library_name = "openlineage-airflow"
+    check_library_version(library_name, get_latest_version_pypi(library_name))
+    from openlineage.airflow.adapter import OpenLineageAdapter
+    from openlineage.client.transport.http import HttpConfig
+    from openlineage.client.transport.kafka import KafkaConfig
+    from openlineage.client.transport.file import FileConfig
+    from openlineage.client.transport.console import ConsoleConfig
+    try:
+        config = OpenLineageAdapter().client.transport.config
+        config_type = type(config)
+        log.debug(f"OpenLineage config: {config}")
+        kwargs['ti'].xcom_push(key='OL_type', value='OL_library')
+    except Exception as e:
+        raise Exception(
+            "OpenLineage config is empty, please setup OpenLineage using https://openlineage.io/docs/client/python - ",
+            e)
+
+    if config_type is HttpConfig:
+        return 'http_config'
+    elif config_type is KafkaConfig:
+        return 'kafka_config'
+    elif config_type is FileConfig:
+        return 'file_config'
+    elif config_type is ConsoleConfig:
+        return 'console_config'
+
+
+def provider_package_config(**kwargs):
+    """
+    This function is used to check the OpenLineage configuration when using provider package.
+    This function also ensures that the OpenLineage provider package is installed and up-to-date.
+    pushes the OL_type to xcom for other tasks to refer.
+    returns the branch based on the config type.
+    :param kwargs:
+    """
+    log.info("Airflow version is >=2.7.0, checking OpenLineage provider package configuration")
+    library_name = "apache-airflow-providers-openlineage"
+    check_library_version(library_name, get_latest_version_pypi(library_name))
+
+    from airflow.providers.openlineage.plugins.listener import get_openlineage_listener
+    config = get_openlineage_listener().adapter.get_openlineage_config()
+    log.debug(f"Config: {config}")
+    if config is None:
+        raise Exception(
+            "OpenLineage provider package config is empty, please setup OpenLineage using https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/guides/user.html")
+
+    config_type = config['type']
+    kwargs['ti'].xcom_push(key='OL_type', value='provider_package')
+    if config_type == 'http':
+        return 'http_config'
+    elif config_type == 'kafka':
+        return 'kafka_config'
+    elif config_type == 'file':
+        return 'file_config'
+    elif config_type == 'console':
+        return 'console_config'
+
+
+def http_config(**kwargs):
+    """
+    This function is used to check the HttpConfig for OpenLineage events ingestion.
+    extracts and pushes the url to xcom for other tasks to refer.
+    returns the branch based on the backend type. (marquez or other)
+    :param kwargs:
+    """
+    log.info("Config type is HttpConfig")
+    OL_type = kwargs['ti'].xcom_pull(key='OL_type', task_ids='OL_library')
```

Comment:
inside the function, we should not use uppercase variables

Reply:
updated